src/test/java/org/osjava/sj/SimpleJndiNewTest.java

```diff
@@ -338,6 +340,55 @@ public void beanSetterNotSharedMixedTypes() throws Exception {
                 ctx1.close();
             }
         }
+        */
+        // TODO remove, obsolete by beanWithSupportedSetters
+    }
+
+    @Test
+    public void beanWithSupportedSetters() throws Exception {
+        InitialContext ctx1 = null;
+        try {
+            final Hashtable<String, String> env = new Hashtable<String, String>();
+            env.put("org.osjava.sj.root",
+                    "file://src/test/resources/roots/beanWithSupportedSetters");
+            env.put("java.naming.factory.initial", "org.osjava.sj.SimpleContextFactory");
+            env.put("org.osjava.sj.delimiter", "/");
+            env.put("org.osjava.sj.space", "java:comp/env");
+            ctx1 = new InitialContext(env);
+            final BeanWithSupportedSetters bean = (BeanWithSupportedSetters) ctx1.lookup("java:comp/env/bean");
+            assert bean != null;
+            assertEquals("Hello World", bean.getString());
+            assertEquals("Hello World", bean.getCharSequence());
+            assertEquals(true, bean.getBooleanPrimitive());
+            assertEquals(false, bean.getBooleanObject());
+            assertEquals(Byte.parseByte("5"), bean.getBytePrimitive());
+            assertEquals(new Byte("7"), bean.getByteObject());
+            assertEquals('x', bean.getCharacterPrimitive());
+            assertEquals(new Character('Z'), bean.getCharacterObject());
+            assertEquals(Short.parseShort("10"), bean.getShortPrimitive());
+            assertEquals(new Short("11"), bean.getShortObject());
+            assertEquals(Integer.parseInt("100"), bean.getIntegerPrimitive());
+            assertEquals(new Integer("101"), bean.getIntegerObject());
+            assertEquals(Long.parseLong("1000"), bean.getLongPrimitive());
+            assertEquals(new Long("1001"), bean.getLongObject());
+            assertEquals(Float.parseFloat("2000"), bean.getFloatPrimitive());
```

Comment:
The two argument constructor is deprecated and fails in my case with
java.lang.AssertionError: Use assertEquals(expected, actual, delta) to compare floating-point numbers

	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertEquals(Assert.java:667)
	at org.junit.Assert.assertEquals(Assert.java:656)
	at org.osjava.sj.SimpleJndiNewTest.beanWithSupportedSetters(SimpleJndiNewTest.java:374)

Reply:
On 3. Dec 2017, at 03:42, jwaterloo <notifications@github.com> wrote:

"Feel free to make the third parameter 0.0f on 372 and 0.0 on
374."